File: src/main/java/social/alone/server/link/LinkService.java
package social.alone.server.link;


import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import social.alone.server.event.Event;

import javax.transaction.Transactional;

@Service
@Transactional
@RequiredArgsConstructor
public class LinkService {

  private final LinkRepository linkRepository;

  public Link createLink(Event event){
    Link link = event.createLink();
    return linkRepository.save(link);
  }
}

File: src/main/kotlin/social/alone/server/link/LinkService.kt
package social.alone.server.link


import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import social.alone.server.event.domain.Event

@Service
@Transactional
class LinkService(val linkRepository: LinkRepository) {

    fun createLink(event: Event): Link {
        val link = event.createLink()
        return linkRepository.save(link)
    }
}
